README.md

```diff
@@ -16,32 +16,46 @@ This assumes that you have `Go` installed and setup.
 
 `pip3` and `python3` is what we need for development purpose.
 
-All dependecies are added in `dep` and we need to make sure we install `fresh` although it is already added to the lock.
 
+### How to install
 
-#### Development
-
-This is a service and it doesn't have any `UI` so in order to use this project or hack on the project we need a way to communicate with it. We can use the good old `curl` but let's be a little adventurous and use [POSTMAN](https://www.getpostman.com/) for the same.
-
-Make sure to carry on the following steps:
+1. [Install Go](https://golang.org/doc/install)
+2. Run the following commands
 
 ```bash
+$ make all
+$ dexer
+```
 
-make clean # for cleaning up
-make dev # for development purposes
-make logs # for getting the dummy directory ready
+Open http://localhost:8000 in your browser to open the user interface.
 
+The default configurations can be changed by editing the `config.json` file.
+```
+{
+    "RootDirectory": "logs",
+    "IndexFilename": "irclogs.bleve",
+    "Port": ":8000"
+}
 ```
 
-Once you did this you will notice that there is a `logs` directory created with different folder and each folder will have different files. Remember all these files are dummy files and they have some random junk.
+### API
 
-Then run `make build` this should start the server. Once everything is working fine install the postman plugin for your browser. And from that plugin you need to hit the endpoint as:
+Once everything is working fine install the postman plugin for your browser. And from that plugin you need to hit the endpoint as:
 
 `localhost:8000/search/american`
 
-Here `american` is the query word that I passed, make sure to open any file in the `logs/` directory and find a word to search. It will look like this:
+Here american is the query word that I passed, make sure to open any file in the logs/ directory and find a word to search. It will look like this:
 
 ![Missing screenshot](https://image.ibb.co/cWP5TA/postman-query.png "Postman Screenshot")
 
 
-This also has the response `file-indexer` sends back.
+### Run locally using docker
+
+You can run the application using Docker in your local machine. It will use the `Dockerfile` instructions. Make sure you have [Docker](https://www.docker.com/) installed in your machine.
+
+Run the following commands to build and run the docker image.
+
+```bash
+$ make docker-build # builds docker image
+$ make docker-run # runs the image in new container
+```
```

Comment:
Add a section for development, write about how to generate a temporary folder with dummy data and test it